### PR TITLE
feat: theme for design system [ME-4243]

### PR DESCRIPTION
**Pull request by shubhnik**
Inbuilt and custom themes can be injected to the Design System.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "node": ">=10.18.0"
   },
   "dependencies": {
+    "lodash.merge": "^4.6.2",
```
[the remark this answers is not in the document]
Comment (shubhnik) [reply package.json]: Was using it for createTheme, anyways we are removing it. 👍

File: src/tokens/theme.native.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { fonts, colors } from '../tokens';
+import { lightTheme as baseLightTheme, darkTheme as baseDarkTheme } from './theme.web';
+
+// For  styled components in React Native: converts unitless numbers to "px" strings
+const convertToPxString = (source) =>
```
Comment (kamleshchandnani) [inline src/tokens/theme.native.js]: rename `convertToPxString` to `getPxValue`
rename `source` to `value`

Also this is a generic function so it should just accept a value and return the px value the reduce should not be part of this function

File: src/tokens/theme.native.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  Object.keys(source).reduce((acc, current) => ({ ...acc, [current]: `${source[current]}px` }), {});
+
+const nativeCompatibleFontProperties = {
```
Comment (kamleshchandnani) [inline src/tokens/theme.native.js]: rename `nativeCompatibleFontProperties` to `nativeFonts`

File: storybook/native/main.js [hunk 4]
```diff
@@ -3,10 +3,9 @@ import { AppRegistry } from 'react-native';
 import { getStorybookUI, configure, addDecorator } from '@storybook/react-native';
 import styled, { ThemeProvider } from 'styled-components';
 
+import theme from '../../src/tokens/theme.native';
```
Comment (kamleshchandnani) [inline storybook/native/main.js]: you don't need `theme.native` metro will figure out on it's own. Just write 
`import theme from '../../src/tokens/theme';`

File: src/tokens/theme.native.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  ...baseLightTheme,
+  fonts: nativeFonts,
+};
+
+const darkTheme = {
+  ...baseDarkTheme,
+  fonts: nativeFonts,
+};
+
+const theme = { ...lightTheme, ...colors };
```
Comment (deorukhkarchaitanya) [inline src/tokens/theme.native.js]: Why are you adding all the colors into theme?

Comment (shubhnik) [reply src/tokens/theme.native.js]: colors goes into default `theme` export and not in `lightTheme` and `darkTheme`.

Comment (deorukhkarchaitanya) [reply src/tokens/theme.native.js]: I thought we concluded that default theme == light theme

File: src/tokens/theme.native.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { fonts, colors } from '../tokens';
+import { lightTheme as baseLightTheme, darkTheme as baseDarkTheme } from './theme.web';
+
+// For  styled components in React Native: converts unitless numbers to "px" strings
+const getPxValue = (value) => `${value}px`;
+
+const convertToPx = (property) =>
```
Comment (kamleshchandnani) [inline src/tokens/theme.native.js]: why can't you write this reduce directly in front of `lineHeight` and `letterSpacing`? That's what the comment was for. Adding these tiny things in a function is unnecessary optimisation

File: src/tokens/theme.native.js (hunk 5 above)
Comment (kamleshchandnani) [reply src/tokens/theme.native.js]: @shubhnik We need to just say `const theme = lightTheme`. That's what we all concluded yesterday

File: src/tokens/theme.native.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { fonts, colors } from '../tokens';
```
Comment (kamleshchandnani) [inline src/tokens/theme.native.js]: Where's `spacings`? That is also part of your theme

File: src/tokens/theme.web.js [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        400: colors.slate[400],
+      },
+      darkBlue: {
+        200: colors.darkBlue[200],
+      },
+    },
+  },
+};
+
+const theme = { ...lightTheme, ...colors };
```
Comment (kamleshchandnani) [inline src/tokens/theme.web.js]: `const theme = lightTheme`

File: src/tokens/theme.web.js [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+      300: colors.green[300],
+      600: colors.green[600],
+      900: colors.green[900],
+    },
+    neutral: {
+      300: colors.honey[300],
+      600: colors.honey[600],
+      900: colors.honey[900],
+    },
+    exception: {
```
Comment (kamleshchandnani) [inline src/tokens/theme.web.js]: remove this. `exception` will break. It doesn't follow our structure. I've discussed with the design team

File: src/tokens/theme.web.js [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+        400: colors.slate[400],
+      },
+      darkBlue: {
+        200: colors.darkBlue[200],
+      },
+    },
+  },
+};
+
+const theme = { ...lightTheme, colors: { ...lightTheme.colors, ...colors } };
```
Comment (kamleshchandnani) [inline src/tokens/theme.web.js]:
```js
colors: { ...colors, ...lightTheme.colors }
```
All the overrides properties should come last

File: src/tokens/theme.native.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+import { lightTheme as baseLightTheme, darkTheme as baseDarkTheme } from './theme.web';
+
+// For  styled components in React Native: converts unitless numbers to "px" strings
+const getPxValue = (value) => `${value}px`;
+
+const nativeFonts = {
+  ...fonts,
+  lineHeight: Object.keys(fonts.lineHeight).reduce(
+    (acc, val) => ({ ...acc, [val]: getPxValue(fonts.lineHeight[val]) }),
```
Comment (kamleshchandnani) [inline src/tokens/theme.native.js]: `value`. please avoid shortnames

**Pull request merged**